subprojects/core/src/integTest/groovy/org/gradle/api/tasks/MissingTaskDependenciesIntegrationTest.groovy

```diff
@@ -223,7 +217,82 @@ class MissingTaskDependenciesIntegrationTest extends AbstractIntegrationSpec {
         """
 
         expect:
-        executer.expectDocumentedDeprecationWarning(DEPRECATION_WARNING)
+        expectMissingDependencyDeprecation(":producer", ":consumer")
         succeeds("producer", "consumer")
     }
+
+    def "can have tasks which consume a filtered project directory"() {
+        file("src/main/java/MyClass.java").createFile()
+        buildFile << """
+            task firstTask {
+                def outputFile = file("build/output1.txt")
+                outputs.file(outputFile)
+                doLast {
+                    outputFile.text = "first"
+                }
+            }
+            task lastTask {
+                def outputFile = file("build/output2.txt")
+                outputs.file(outputFile)
+                doLast {
+                    outputFile.text = "last"
+                }
+            }
+            task packageProjectDir(type: Zip) {
+                from(project.projectDir) {
+                    include 'src/**'
+                }
+                destinationDirectory = file("build")
+                archiveBaseName = "output3"
+            }
+        """
+
+        when:
+        run("firstTask", "packageProjectDir", "lastTask")
+        then:
+        executedAndNotSkipped(":firstTask", ":packageProjectDir", ":lastTask")
+    }
+
+    def "detects filtered trees with missing dependencies"() {
+        file("src/main/java/MyClass.java").createFile()
+        buildFile << """
+            task firstTask {
+                def outputFile = file("build/output1.txt")
+                outputs.file(outputFile)
+                doLast {
+                    outputFile.text = "first"
+                }
+            }
+            task lastTask {
+                def outputFile = file("build/output2.txt")
+                outputs.file(outputFile)
+                doLast {
+                    outputFile.text = "last"
+                }
+            }
+            task packageProjectDir(type: Zip) {
+                from(project.projectDir) {
+                    include 'build/**'
+                }
+                destinationDirectory = file("build")
```

Comment:
This is a bit misleading for me. It would be easier for me to understand the goal here if the stuff that is supposed to collide was in one directory (`build/stuff-to-zip` or `build/problematic-output` or something) while the zip itself could be created in `build`.

Reply:
I changed the test.